### DISCUSSION TITLE
Tandems: What are they?

### DISCUSSION THREAD

**Discussion by ghost** [scenerystack/community#40] [Q&A]
Hello Everyone,

I have noticed that that the term "tandem" appears countless times in PHET sim code. I was wondering if anyone could explain what they are, their use, and how to implement them to me?

Thanks,
Rex

**Comment by samreid** [comment]
Rex, great question!  Scenerystack provides a way of providing unique identifiers to different instances as they are created. There is a type named `Tandem.ts` that supports this: https://github.com/phetsims/tandem/blob/main/js/Tandem.ts

These are used by PhET Interactive Simulations to create the PhET-iO simulations. For other use cases, I believe they can be ignored, or removed.

**Reply by ghost** [reply]
So, 

Just to confirm, a tandem is basically a label for an instance of an class?

Thanks,
Rex
